sdk/python/kfp/compiler/compiler_test.py

```diff
@@ -4258,6 +4258,65 @@ def test_workspace_config_validation(self):
         self.assertIn('required and cannot be empty', str(context.exception))
 
 
+class TestPipelineSemaphoreMutex(unittest.TestCase):
+
+    def test_pipeline_with_semaphore(self):
+        """Test that pipeline config correctly sets the semaphore key."""
+        config = PipelineConfig()
+        config.semaphore_key = 'semaphore'
+
+        @dsl.pipeline(pipeline_config=config)
+        def my_pipeline():
+            task = comp()
+
+        with tempfile.TemporaryDirectory() as tempdir:
+            output_yaml = os.path.join(tempdir, 'pipeline.yaml')
+            compiler.Compiler().compile(
+                pipeline_func=my_pipeline, package_path=output_yaml)
+
+            with open(output_yaml, 'r') as f:
+                pipeline_docs = list(yaml.safe_load_all(f))
+
+        pipeline_spec = None
+        for doc in pipeline_docs:
+            if 'platforms' in doc:
+                pipeline_spec = doc
+                break
+
+        if pipeline_spec:
+            kubernetes_spec = pipeline_spec['platforms']['kubernetes'][
+                'pipelineConfig']
+            self.assertEqual(kubernetes_spec['semaphoreKey'], 'semaphore')
+
+    def test_pipeline_with_mutex(self):
+        """Test that pipeline config correctly sets the mutex name."""
+        config = PipelineConfig()
+        config.mutex_name = 'mutex'
+
+        @dsl.pipeline(pipeline_config=config)
+        def my_pipeline():
+            task = comp()
+
+        with tempfile.TemporaryDirectory() as tempdir:
+            output_yaml = os.path.join(tempdir, 'pipeline.yaml')
+            compiler.Compiler().compile(
+                pipeline_func=my_pipeline, package_path=output_yaml)
+
+            with open(output_yaml, 'r') as f:
+                pipeline_docs = list(yaml.safe_load_all(f))
+
+        pipeline_spec = None
+        for doc in pipeline_docs:
+            if 'platforms' in doc:
+                pipeline_spec = doc
+                break
+
+        if pipeline_spec:
```

Comment:
Same comment as the previous test

Reply:
Done, replaced with an assertion and the platform_spec var name.